dlp/api/Snippets/DeidentifyWithDateShift.cs

```diff
@@ -0,0 +1,141 @@
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// [START dlp_deidentify_date_shift]
+
+using Google.Cloud.Dlp.V2;
+using Google.Protobuf;
+using System;
+using System.IO;
+using System.Linq;
+
+public class DeidentifyWithDateShift
+{
+    public static DeidentifyContentResponse Deidentify(
+        string projectId,
+        string inputCsvFilePath,
+        int lowerBoundDays,
+        int upperBoundDays,
+        string dateFields,
+        string contextField = "",
+        string keyName = "",
+        string wrappedKey = "")
+    {
+        var dlp = DlpServiceClient.Create();
+
+        // Read file
+        var csvLines = File.ReadAllLines(inputCsvFilePath);
+        var csvHeaders = csvLines[0].Split(',');
+        var csvRows = csvLines.Skip(1).ToArray();
+
+        // Convert dates to protobuf format, and everything else to a string
+        var protoHeaders = csvHeaders.Select(header => new FieldId { Name = header });
+        var protoRows = csvRows.Select(CsvRow =>
+        {
+            var rowValues = CsvRow.Split(',');
+            var protoValues = rowValues.Select(RowValue =>
+            {
+                if (System.DateTime.TryParse(RowValue, out var parsedDate))
+                {
+                    return new Value
+                    {
+                        DateValue = new Google.Type.Date
+                        {
+                            Year = parsedDate.Year,
+                            Month = parsedDate.Month,
+                            Day = parsedDate.Day
+                        }
+                    };
+                }
+                else
+                {
+                    return new Value
+                    {
+                        StringValue = RowValue
+                    };
+                }
+            });
+
+            var rowObject = new Table.Types.Row();
+            rowObject.Values.Add(protoValues);
+            return rowObject;
+        });
+
+        var dateFieldList = dateFields
+            .Split(',')
+            .Select(field => new FieldId { Name = field });
+
+        // Construct + execute the request
+        var dateShiftConfig = new DateShiftConfig
+        {
+            LowerBoundDays = lowerBoundDays,
+            UpperBoundDays = upperBoundDays
+        };
+        var hasKeyName = !string.IsNullOrEmpty(keyName);
+        var hasWrappedKey = !string.IsNullOrEmpty(wrappedKey);
+        var hasContext = !string.IsNullOrEmpty(contextField);
+        if (hasKeyName && hasWrappedKey && hasContext)
```

Comment:
For simplicity, can we assume that these will be present always?

Reply:
I'm not sure what the implications of this being in an incorrect state are, so probably makes sense to leave it as-is for now.